IPython/core/completer.py

```diff
@@ -124,6 +123,7 @@
 import string
 import warnings
 
+from collections import defaultdict
```

Comment:
Could you help me and tell me what is this that you are importing?

Reply:
`defaultdict` is a modified version of `dict` available in the default `collections` package.
It allows to specify a default value that is assigned to each key that doesn't exist.

```
a = defaultdict(lambda: True)
assert a["hello"] == True
```